### PR TITLE
[style] Use clang-format to sort includes

### LINKED ISSUE #9482: [style] Use clang-format to sort includes
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

Before the commit #9305 , if we use `clang-format` to format our codebase, the `clang-format` sorts the includes.

#9305 claims that there are some compilation errors occurring after we sort the includes in our codebase. Generally speaking, it is right due to the conditional includes (some includes must be put after others, see [google code style](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes)). However, the number of these exceptions is small, we can treat them case by case.

There are some approaches to handle these exceptions:
1. Exclude the whole file - Adding it to `.clang-format-ignore`
2. Exclude some regions in the file - Please refer to [Clang-Format Style Options](https://clang.llvm.org/docs/ClangFormatStyleOptions.html#disabling-formatting-on-a-piece-of-code)

All in all, it is very helpful if we can unify the order of includes. After investigation, the compilation errors was caused by `be/src/service/brpc.h`. In `brpc.h`, the `service/brpc_conflict.h` must be included first.

### Solution

1. Revert the setting of `.clang-format`
2. Fix the compilation errors.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by adonis0147**
# Proposed changes

Issue Number: close #9482 

## Problem Summary:

Please refer to #9482 .

## Checklist(Required)

1. Does it affect the original behavior: No
2. Has unit tests been added: No Need
3. Has document been added or modified: No Need
4. Does it need to update dependencies: No
5. Are there any changes that cannot be rolled back: No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by adonis0147** [comment]
CC @chenlinzhong @morningman, please refer to  issue: #9482 .

**Comment by yiguolei** [comment]
@adonis0147  I am just curious, is there any tool to find unused includes?

File: be/src/common/status.h [hunk 1]
```diff
@@ -5,13 +5,13 @@
 #pragma once
 
 #include <fmt/format.h>
+#include <glog/logging.h>
+
+#include <boost/stacktrace.hpp>
```
Comment (yiguolei) [inline be/src/common/status.h:10]: why boost is behind glog? Does the format not sort includes according to lexical order？

**Comment by adonis0147** [comment]
Hi @yiguolei , the answer is yes. Currently, I use the `clangd` (version >= 14.0.0) to detect the unused includes (please refer to #9117). However, #9117 isn't a standalone command solution.

I tried the tool [include-what-you-use](https://include-what-you-use.org/) before, but I failed to apply the tool to our codebase.

File: be/src/common/status.h (hunk 1 above)
Comment (adonis0147) [reply be/src/common/status.h:10]: Our clang-format style is based on [Google Code Style](https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes) (`BasedOnStyle: Google` in `.clang-format`). The rules doesn't apply lexical order simply.

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by chenlinzhong** [comment]
LGTM

**Pull request merged**